### DISCUSSION TITLE
Flutter android build is passing locally but failing on codemagic

### DISCUSSION THREAD

**Discussion by jamespainted** [codemagic-ci-cd/codemagic-docs#2026] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Run `flutter build appbundle --release` locally and on run workflow on codemagic

### Expected results

Expect successful build both locally and on codemagic

### Actual results

Instead got this on codemagic
`Found latest version code from "internal" track: 2

Upgrading build.gradle
Running Gradle task 'bundleRelease'...                          
../.pub-cache/hosted/pub.dev/flutter_quill-7.3.2/lib/src/widgets/raw_editor.dart:283:7: Error: The non-abstract class 'RawEditorState' is missing implementations for these members:
 - TextSelectionDelegate.liveTextInputEnabled
Try to either
 - provide an implementation,
 - inherit an implementation from a superclass or mixin,
 - mark the class as abstract, or
 - provide a 'noSuchMethod' implementation.

class RawEditorState extends EditorState
      ^^^^^^^^^^^^^^
../programs/flutter/packages/flutter/lib/src/services/text_input.dart:1058:12: Context: 'TextSelectionDelegate.liveTextInputEnabled' is defined here.
  bool get liveTextInputEnabled => false;
           ^^^^^^^^^^^^^^^^^^^^
../.pub-cache/hosted/pub.dev/flutter_quill-7.3.2/lib/src/widgets/raw_editor.dart:351:47: Error: Required named parameter 'onLiveTextInput' must be provided.
    return EditableText.getEditableButtonItems(
                                              ^
../programs/flutter/packages/flutter/lib/src/widgets/editable_text.dart:1832:38: Context: Found this candidate, but the arguments don't match.
  static List<ContextMenuButtonItem> getEditableButtonItems({
                                     ^^^^^^^^^^^^^^^^^^^^^^
Target kernel_snapshot failed: Exception

FAILURE: Build failed with an exception.

* Where:
Script '/Users/builder/programs/flutter/packages/flutter_tools/gradle/src/main/groovy/flutter.groovy' line: 1297

* What went wrong:
Execution failed for task ':app:compileFlutterBuildRelease'.
> Process 'command '/Users/builder/programs/flutter/bin/flutter'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1m 7s
Running Gradle task 'bundleRelease'...                             68.6s
Gradle task bundleRelease failed with exit code 1

Build failed :|
Step 9 script `Build AAB with Flutter` exited with status code 1`

### Build id (optional)
64f41e922b1bcc018bf9256c

_No response_

**Comment by himesh-cm** [comment, marked as answer]
Hi @jamespainted, it looks like a known issue with the plugin as per this [GH](https://github.com/singerdmx/flutter-quill/issues/1384) issue. Workaround is to `use version below 7.4.0`.
